### PR TITLE
Add Valheim Plus Mod + Readme Fixes

### PR DESCRIPTION
**Pull request by gOOvER**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [ ] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

### New Server Submissions:

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your server use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the server to the main README.md?
4. [x] Have you added a unique README.md for the server you are adding?

### Changes to an existing Egg:

1. [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
2. [ ] Have you tested your Egg changes?

### CODE REVIEW

File: steamcmd_servers/valheim/valheim_plus/egg-valheim-plus-mod.json [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    },
+    "exported_at": "2021-02-23T15:36:16+01:00",
+    "name": "Valheim Plus Mod",
+    "author": "info@goover.de",
+    "description": "A brutal exploration and survival game for 1-10 players, set in a procedurally-generated purgatory inspired by viking culture.",
+    "features": null,
+    "images": [
+        "quay.io\/parkervcp\/pterodactyl-images:debian_source"
+    ],
+    "startup": ".\/start_server_bepinex.sh",
```
Comment (parkervcp) [inline steamcmd_servers/valheim/valheim_plus/egg-valheim-plus-mod.json:15]: I am against using shell scripts to start things.

Comment (gOOvER) [reply steamcmd_servers/valheim/valheim_plus/egg-valheim-plus-mod.json:15]: > I am against using shell scripts to start things.

Thats the normal Startscript from the Modloader. is installed with the mod.

Comment (gOOvER) [reply steamcmd_servers/valheim/valheim_plus/egg-valheim-plus-mod.json:15]: LSGM use the same:

https://github.com/valheimPlus/ValheimPlus/issues/49#issuecomment-784275395

**Comment by MatusBoa** [comment]
FYI:
Recently I tried to run Valheim+ on pterodactyl but found out that `start_server_bepinex.sh` / `run_bepinex.sh` are not compatible with flags that are passing into `valheim_server.x86_64` by original egg (https://github.com/parkervcp/eggs/blob/master/steamcmd_servers/valheim/egg-valheim.json#L15).

My solution to this is to modify `run_bepinex.sh` like this

```bash
#!/bin/sh
export templdpath=$LD_LIBRARY_PATH

# BepInEx-specific settings
# NOTE: Do not edit unless you know what you are doing!
####
export DOORSTOP_ENABLE=TRUE
export DOORSTOP_INVOKE_DLL_PATH=./BepInEx/core/BepInEx.Preloader.dll
export DOORSTOP_CORLIB_OVERRIDE_PATH=./unstripped_corlib

export LD_LIBRARY_PATH=./doorstop_libs:$LD_LIBRARY_PATH
export LD_PRELOAD=libdoorstop_x64.so:$LD_PRELOAD
####


export LD_LIBRARY_PATH=./linux64:$LD_LIBRARY_PATH
export SteamAppId=892970

echo "Starting server PRESS CTRL-C to exit"

SERVER_NAME=$1
SERVER_PORT=$2
SERVER_WORLD=$3
SERVER_PASSWORD=$4

# Tip: Make a local copy of this script to avoid it being overwritten by steam.
# NOTE: Minimum password length is 5 characters & Password cant be in the server name.
# NOTE: You need to make sure the ports 2456-2458 is being forwarded to your server through your local router & firewall.
./valheim_server.x86_64 -nographics -batchmode -name $SERVER_NAME -port $SERVER_PORT -world $SERVER_WORLD -password $SERVER_PASSWORD -public 1

export LD_LIBRARY_PATH=$templdpath
```

Startup command now looks like this:

`sh ./run_bepinex.sh {{SERVER_NAME}} {{SERVER_PORT}} "{{WORLD}}" "{{PASSWORD}}" > >(sed -uE "{{FILTER}}") & trap "{{STOP}}" 15; wait $!`

**Comment by gOOvER** [comment]
i tryed this qllready and it don't work.

**Comment by MatusBoa** [comment]
> i tryed this qllready and it don't work.

My solution above works just fine

**Comment by gOOvER** [comment]
i will test. but the startscript will overwritten on every Update.

btw u use the wrong start.sh

**Comment by gOOvER** [comment]
you use a old Version of the startscript

**Comment by MatusBoa** [comment]
Yeah, does not matter. I just wanted to point out that you have to make support for parameters passed from Pterodactyl.

**Comment by gOOvER** [comment]
maybe i create an own one and add it to repro too.

I will test

**Comment by grimesy88** [comment]
Keen for this @gOOvER will you post in your repo?

**Comment by gOOvER** [comment]
> ```shell
> SERVER_NAME=$1
> SERVER_PORT=$2
> SERVER_WORLD=$3
> SERVER_PASSWORD=$4
> ```

So, i'm at home and  i cant see, whats different? I parse the original file, so on every update no need to modifiy it again. 

I will not add your solution, because i don't see any changes. btw: -public 1 was removed ;)

**Comment by grimesy88** [comment]
@gOOvER I am testing this on my nodes now and it seems to work really well. Are there any pitfalls from a hosting point of view I need to be aware of? Updates to the game wont kill the start scripts because those are parsed on each startup right?

**Comment by gOOvER** [comment]
Yes, its parsed on every start.

**Comment by grimesy88** [comment]
Will this egg auto update BepInEx and therefore the Valheim+ too?

**Comment by gOOvER** [comment]
Yes, when u reinstall the server.

**Comment by grimesy88** [comment]
Following todays update Is it possible to add the public variable back in to the start command? I've tried adding a variable and adding it to the parser so it populates the start script file but it ignores it, i must be missing something?

**Comment by gOOvER** [comment]
With the next Release of V+ they change the startscript and add Startup Flagge, so this PR i WIP

**Comment by grimesy88** [comment]
ok i will keep an eye on release ty

**Comment by grimesy88** [comment]
Big update to v+ today has changed the start_server_bepinex script

**Comment by grimesy88** [comment]
FYI this new update as you rightly eluded to before, allows us to pass flags/variables via the start parameters. Confirming that with this egg using ./start_server_bepinex.sh -name "{{SERVER_NAME}}" -port {{SERVER_PORT}} -world "{{WORLD}}" -password "{{PASSWORD}}" for example will start the server without issue.

**Comment by gOOvER** [comment]
I know ;) I testet allready the Alpha :)

**Comment by gOOvER** [comment]
Updated to use latest Startscript :) Ready to merge

**Comment by gOOvER** [comment]
YOU MUST UPDATE TO THE LATEST COMMIT !!!!!!!!!!!!!!! ELSE THE SERVER DONT RUN !!!!!!!!!!!!

**Comment by gOOvER** [comment]
@parkervcp I updated the Egg. Now all things are in startup cmd and script is not longer used

**Comment by grimesy88** [comment]
is the console filter now working on this commit @gOOvER ?

**Comment by gOOvER** [comment]
> is the console filter now working on this commit @gOOvER ?

for me it works :)

File: steamcmd_servers/valheim/valheim_plus/egg-valheim-plus-mod.json (hunk 1 above)
Comment (gOOvER) [reply steamcmd_servers/valheim/valheim_plus/egg-valheim-plus-mod.json]: fixed with latest commit

**Comment by gOOvER** [comment]
> is the console filter now working on this commit @gOOvER ?

Did you test it? can you give me feedback?

**Comment by grimesy88** [comment]
It does indeed seem to work :) ty

**Comment by SG-Brandon** [comment]
Has the startup script changed?  Everyone using linux servers has the version compatibility issue.  Check out what the dude in this last comment said https://www.reddit.com/r/valheim/comments/m4xf3s/valheimplus/

He linked to this script - https://github.com/valheimPlus/ValheimPlus/blob/d552eb1fc7b2bcca379b8c1227b1de22565e1f6a/resources/unix/start_server_bepinex.sh

Said that fixed the compatibility issue people keep having to change the enforce mod thing to avoid.  Unless I'm doing something wrong?  I just start the egg up and people have compatibility issues.

**Comment by gOOvER** [comment]
if u have this error, quickfix is not installed

**Comment by SG-Brandon** [comment]
> if u have this error, quickfix is not installed

What do you mean exactly?  How do I do that?

**Pull request merged**